test/models/test_dynamic_port_binding.rb

```diff
@@ -455,6 +455,42 @@ module Models
                 end
             end
 
+            describe "injecting another instantiation class" do
+                before do
+                    @port = DynamicPortBinding.new(flexmock, flexmock,
+                                                   output: true, port_resolver: nil)
+                end
+                it "raises if klass is not a subclass of BoundOutputReader" do
+                    foo = Class.new
+                    assert_raises ArgumentError do
+                        DynamicPortBinding::BoundOutputReader.new(
+                            "bla", flexmock, @port, klass: foo
+                        )
+                    end
+                end
+
+                it "instantiates using BoundOutputReader when klass is not passed" do
+                    b = DynamicPortBinding::BoundOutputReader.new("bla", flexmock, @port)
+                    result = b.instanciate(flexmock)
+                    assert(
+                        result.instance_of?(Syskit::DynamicPortBinding::BoundOutputReader)
+                    )
+                end
+
+                it "instantiates using the injected class" do
+                    klass = Class.new Syskit::DynamicPortBinding::BoundOutputReader
+
+                    b = DynamicPortBinding::BoundOutputReader.new(
+                        "bla", flexmock, @port, klass: klass
+                    )
+                    result = b.instanciate(flexmock)
+                    assert(
+                        result.kind_of?(Syskit::DynamicPortBinding::BoundOutputReader)
+                    )
```

Comment:
This makes little sense ... if result.kind_of?(klass) and `klass <= Syskit::DynamicPortBinding::BoundOutputReader` then it is inferred.